### ISSUE TITLE
Add support for linkifying more URI schemes

### ISSUE THREAD

**Issue by cketti** [thundernest/k-9#2265]
We currently linkify HTTP URLs and Bitcoin URIs when converting plain text to HTML. It'd be nice to have support for additional URI schemes. See #684.

* `bridge:` - Tor bridge
* `geo:` - Physical location
* `market:` - Link to app in Google Play (previously Android Market)
* `openpgp4fpr:` - OpenPGP key IDs and fingerprints
* `xmpp:` - Identify or interact with entities using XMPP

However, I'm concerned about the performance impact of this. Ideally we'd use something similar to [linkifyjs](https://github.com/SoapBox/linkifyjs) that goes through the text only once to find all URIs.

We might also want to think about linkifying supported URIs in HTML parts. That's bad when e.g. malicious URLs have been left unlinkified intentionally. But I believe in the majority of cases linkifying is what is expected by users.

**Comment by karishnu** [comment]
I will like to give this a try. Can I work on this?

**Comment by cketti** [comment]
@karishnu: This issue is probably not suitable for a beginner.

**Comment by cketti** [comment]
We could add a generic URI ([RFC 3986](https://datatracker.ietf.org/doc/html/rfc3986)) parser and allow the user to specify which URI schemes they want to have linkified.
